jbpm-human-task/jbpm-human-task-workitems/src/main/java/org/jbpm/services/task/wih/AbstractHTWorkItemHandler.java

```diff
@@ -128,16 +128,17 @@ protected Task createTaskBasedOnWorkItemParams(KieSession session, WorkItem work
         
         task.setSubject(comment);
         
-        String priorityString = (String) workItem.getParameter("Priority");
-        int priority = 0;
-        if (priorityString != null) {
+        Object priorityObject = workItem.getParameter("Priority");
+        
+        if (priorityObject instanceof Number) {
+            task.setPriority(((Number) priorityObject).intValue());
+        } else if (priorityObject != null) {
             try {
-                priority = new Integer(priorityString);
+                task.setPriority(Integer.parseInt(priorityObject.toString()));
             } catch (NumberFormatException e) {
                 // do nothing
             }
         }
-        task.setPriority(priority);
```

Comment:
Previously, when Priority was not present (null), it was set up as 0, but here, it wouldn't be and it is not initialized with zero value in the implementation classes, so potentially it won't have priority, breaking backward compatibility, wdyt?

Reply:
You are right. 
For some reason, when I did the change yesterday, I was assuming that priority was an int, not an Integer, in which case will be initialized to 0 anyway.